src/main/java/seedu/address/logic/commands/MarkCommand.java

```diff
@@ -58,26 +62,81 @@ private static Task createMarkedTask(Task task) {
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
         List<Task> lastShownList = model.getFilteredTaskList();
-
-        if (targetIndex.getZeroBased() >= lastShownList.size()) {
-            throw new CommandException(Messages.MESSAGE_INVALID_TASK_DISPLAYED_INDEX);
+        ArrayList<Task> markedTasks = new ArrayList<>();
+
+
+        // to mark each index in the task list individually.
+        for (int i = 0; i < targetIndexes.size(); i++) {
+            if (targetIndexes.get(i).getZeroBased() >= lastShownList.size()) {
+
+                // in the case where the first few inputted indexes are marked successfully, but one of the latter
+                // inputted indexes throw an error. Harmonia informs the user of the index that caused the error and
+                // also informs the user of the tasks that it marked successfully
+                if (markedTasks.size() > 1) {
+                    throw new CommandException("Index " + targetIndexes.get(i).getOneBased() + " :"
+                            + Messages.MESSAGE_INVALID_TASK_DISPLAYED_INDEX + "\n"
+                            + markedTasksToString(markedTasks.subList(0, markedTasks.size() - 1)));
```

Comment:
Should the sublist() call be `subList(0, markedTasks.size())`? See https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/ArrayList.html#subList(int,int)

Reply:
this is meant to be subList(0, markedTasks.size() - 1) because the index that throws the error is the last element included in the markedTasks list, so this removes the index that throws the error from the message returned!!